tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -189,14 +342,31 @@ public:
    const std::string &GetProcessorName() const { return fProcessorName; }
 
    /////////////////////////////////////////////////////////////////////////////
-   /// \brief Get the model used by the processor.
-   const ROOT::RNTupleModel &GetModel() const { return *fModel; }
-
-   /////////////////////////////////////////////////////////////////////////////
-   /// \brief Get a reference to the entry used by the processor.
+   /// \brief Register a new field for reading during processing.
+   ///
+   /// \tparam T Type of the field to register.
+   ///
+   /// \param[in] fieldName Name of the field to register.
    ///
-   /// \return A reference to the entry used by the processor.
-   const ROOT::REntry &GetEntry() const { return *fEntry; }
+   /// \return An RNTupleProcessorOptionalPtr, which provides access to the field's value.
+   template <typename T>
+   RNTupleProcessorOptionalPtr<T> RegisterField(std::string_view fieldName)
```

Comment:
I think I'd rather mimic the view API here, i.e. `GetView()` returning an object that probably derives from `RNTupleViewBase`.